zebra-chain/src/transaction.rs

```diff
@@ -429,32 +443,6 @@ impl Transaction {
         }
     }
 
-    /// Modify the expiry height of this transaction.
-    ///
-    /// # Panics
-    ///
-    /// - if called on a v1 or v2 transaction
-    #[cfg(any(test, feature = "proptest-impl"))]
-    pub fn expiry_height_mut(&mut self) -> &mut block::Height {
```

Comment:
Moving these test-only methods is a nice refactor.

Nitpick/Optional: We could defer it to try minimizing merge conflicts with the ZSA/NSM changes, but the conflicts would likely be easy to resolve if you'd like to keep it.